allennlp/sanity_checks/task_checklists/utils.py

```diff
@@ -0,0 +1,83 @@
+import string
+from typing import Dict, Callable
+import numpy as np
+
+
+def spacy_wrap(fn: Callable, language: str = "en_core_web_sm", **kwargs):
+    """
+    Wrap the function so that it runs the input text data
+    through a spacy model before the function call.
+    """
+    from allennlp.common.util import get_spacy_model
+    import spacy
+
+    def new_fn(data):
+        if not isinstance(data, spacy.tokens.doc.Doc):
+            model = get_spacy_model(language, **kwargs)
+            if isinstance(data, Dict):
+                for key, val in data.items():
+                    if isinstance(val, str):
+                        data[key] = model(val)
+            elif isinstance(data, str):
+                data = model(data)
+            else:
+                pass
+        return fn(data)
+
+    return new_fn
+
+
+def strip_punctuation(data: str):
+    """
+    Removes all punctuation from the string `data`.
+    """
+    while len(data) and data[-1] in string.punctuation:
+        data = data[:-1]
+    return str(data)
+
+
+def toggle_punctuation(data: str):
+    """
+    If `data` contains any punctuation, it is removed.
+    Otherwise, a `.` is added to the string.
+    Returns a list of strings.
+    """
+    s = strip_punctuation(data)
+    ret = []
+    if s != data:
+        ret.append(s)
+    if s + "." != data:
+        ret.append(s + ".")
+    return ret
```

Comment:
Why not just `return [s]` or `return [s + "."]`?

Reply:
Here's what it's doing: 

`data` = "This was great!"
 Returns ["This was great", "This was great."]

`data` = "The movie was good"
Returns ["The movie was good."]